hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/ftp/FTPFileSystem.java

```diff
@@ -110,7 +111,9 @@ public void initialize(URI uri, Configuration conf) throws IOException { // get
 
     // get port information from uri, (overrides info in conf)
     int port = uri.getPort();
-    port = (port == -1) ? FTP.DEFAULT_PORT : port;
+    if(port == -1){
+      port = conf.getInt(FS_FTP_HOST_PORT, FTP.DEFAULT_PORT);
```

Comment:
assuming we have documentation for the FTP connector, you are going to have to document this new option.

Reply:
indeed, but unfortunately there is no documentation for this connector at the moment.

Reply:
that's got you out of trouble :)